transport/transport.go

```diff
@@ -341,8 +341,10 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 type ConnectOptions struct {
 	// UserAgent is the application user agent.
 	UserAgent string
-	// Dialer specifies how to dial a network address.
-	Dialer func(string, time.Duration) (net.Conn, error)
+	// Cancel is closed to indicate that dialing should be cancelled.
+	Cancel chan struct{}
+	// dialer specifies how to dial a network address.
+	dialer func(string, time.Duration, <-chan struct{}) (net.Conn, error)
```

Comment:
Not really true - the other user is the end2end test. That's a real user, isn't it? :trollface: